src/ScrollToCell.tsx

```diff
@@ -0,0 +1,54 @@
+import { useRef } from 'react';
+
+import { useLayoutEffect } from './hooks';
+import { scrollIntoView } from './utils';
+
+export interface PartialPosition {
+  readonly idx?: number | undefined;
+  readonly rowIdx?: number | undefined;
+}
+
+export default function ScrollToCell({
+  scrollToPosition: { idx, rowIdx },
+  gridElement,
+  setScrollToCellPosition
+}: {
+  scrollToPosition: PartialPosition;
+  gridElement: HTMLDivElement;
+  setScrollToCellPosition: (cell: null) => void;
+}) {
+  const ref = useRef<HTMLDivElement>(null);
+
+  useLayoutEffect(() => {
+    function removeScrollToCell() {
+      setScrollToCellPosition(null);
+    }
+
+    const observer = new IntersectionObserver(removeScrollToCell, {
```

Comment:
Won't the callback be called when the element's visibility goes below the threshold too?
Not sure if it can be an issue.

What happens when the cell is already visible?

Reply:
> What happens when the cell is already visible?

It correctly calls the callback and remove the cell.

Not sure about the other question. I will check

Reply:
May be we can check `intersectionRatio`

Reply:
But like you said, it should not be an issue